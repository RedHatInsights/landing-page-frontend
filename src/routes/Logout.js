import React from 'react';
import { Title, Button } from '@patternfly/react-core';

import './Logout.scss';

const onButtonClick = (url) => {
    window.location.href = `./${url}`;
};

const Logout = () => {
    return (
        <section className="pf-l-page__main-section pf-c-page__main-section ins-c-page__logout">
            <Title headingLevel='h1' size='3xl'> You have logged out. </Title>
            <Button variant='link' onClick={ () => onButtonClick('') }>Take me home</Button>
        </section>
    );
};

export default Logout;
