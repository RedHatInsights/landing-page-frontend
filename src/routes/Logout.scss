@import '~@red-hat-insights/insights-frontend-components/Utilities/_mixins';

.ins-c-page__logout {
    justify-content: center;
    flex-direction: column;
    text-align: center;

    h1 { @include rem('margin-bottom', 30px); }
    button { @include rem('font-size', 20px); }
}