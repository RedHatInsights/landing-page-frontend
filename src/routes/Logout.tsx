import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
import { Title } from '@patternfly/react-core/dist/dynamic/components/Title';
import React from 'react';

import './Logout.scss';

const onButtonClick = (url: string) => {
  window.location.href = `./${url}`;
};

const Logout = () => (
  <section className="pf-v5-l-page__main-section pf-v5-c-page__main-section land-c-page__logout">
    <Title headingLevel="h1" size="3xl">
      You have logged out.
    </Title>
    <Button variant="link" onClick={() => onButtonClick('')}>
      Take me home
    </Button>
  </section>
);

export default Logout;
