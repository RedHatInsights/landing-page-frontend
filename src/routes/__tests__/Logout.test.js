import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logout from '../Logout';

describe('Logout component', () => {
    it('should render correctly', () => {
        const wrapper = mount(<Logout />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('click: take me home', () => {
        const wrapper = mount(<Logout />);
        wrapper.find('button').simulate('click');
        expect(window.location.pathname).toBe('/');
    });
});
