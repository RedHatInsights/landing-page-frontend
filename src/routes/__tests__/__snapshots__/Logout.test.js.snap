// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component should render correctly 1`] = `
<Logout>
  <section
    className="pf-l-page__main-section pf-c-page__main-section ins-c-page__logout"
  >
    <Title
      headingLevel="h1"
      size="3xl"
    >
      <h1
        className="pf-c-title pf-m-3xl"
      >
        You have logged out.
      </h1>
    </Title>
    <Button
      onClick={[Function]}
      variant="link"
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link"
        data-ouia-component-id="OUIA-Generated-Button-link-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button"
      >
        Take me home
      </button>
    </Button>
  </section>
</Logout>
`;
